### ISSUE TITLE
components-return-once: rule triggered if a function declaration is after the return

### ISSUE THREAD

**Issue by derekrjones** [solidjs-community/eslint-plugin-solid#163]
components-return-once rule triggered if a function declaration is after the return

```js
export function Comp() {
  //...
  return <div>...</div>;

  function foo() {/*...*/}
}
```

it gives the follow lint warning/error: `Solid components run once, so an early return breaks reactivity. Move the condition inside a JSX element, such as a fragment or <Show />.`

**Expected behavior**

function declarations after a single return should not trigger the rule

**Comment by derekrjones** [comment]
@joshwilsonvu Awesome thanks!

**Comment by joshwilsonvu** [comment]
No problem, great suggestion! Never thought of that use case before.
